### PR TITLE
sys-kernel/raspberrypi-image: update

### PR DESCRIPTION
**Pull request by andrey-utkin**
Replay of PR https://github.com/gentoo/gentoo/pull/11382 with enhancements:
* small well-commented commits,
* unified ebuild for live and non-live cases,
* fixed the pain of cloning full git repo full of blobs in case of installing from 9999.

### CODE REVIEW

**Comment by andrey-utkin** [comment]
@ConiKost please review.

**Comment by andrey-utkin** [comment]
Merged as e6782c31ff831333af4e03fd122d5a163d2afaeb..dcb6a89b72ec33acc8c97e3e844bd7a8bb68177d.

File: sys-kernel/raspberrypi-image/raspberrypi-image-4.14.98_p20190215.ebuild [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+EAPI=7
+
+inherit mount-boot
+
+DESCRIPTION="Raspberry Pi (all versions) kernel and modules"
+HOMEPAGE="https://github.com/raspberrypi/firmware"
+LICENSE="GPL-2 raspberrypi-videocore-bin"
+SLOT="0"
+RESTRICT="binchecks strip"
```
Comment (ConiKost) [inline sys-kernel/raspberrypi-image/raspberrypi-image-4.14.98_p20190215.ebuild]: Why strip? It's allowed and other devs also said on my other packges, that strip doesn't count as "do not modify". I also would drop binchecks, as no QA errors show up.

File: sys-kernel/raspberrypi-image/raspberrypi-image-4.14.98_p20190215.ebuild [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+RDEPEND="sys-boot/raspberrypi-firmware"
+
+if [[ "${PV}" == 9999 ]]; then
```
Comment (ConiKost) [inline sys-kernel/raspberrypi-image/raspberrypi-image-4.14.98_p20190215.ebuild]: What's the purpose in a non 9999 ebuild?

**Comment by andrey-utkin** [comment]
On Sat, Mar 30, 2019 at 05:15:27AM -0700, Conrad Kostecki wrote:
> ConiKost commented on this pull request.
> 
> 
> 
> > +# Copyright 1999-2019 Gentoo Authors
> +# Distributed under the terms of the GNU General Public License v2
> +
> +EAPI=7
> +
> +inherit mount-boot
> +
> +DESCRIPTION="Raspberry Pi (all versions) kernel and modules"
> +HOMEPAGE="https://github.com/raspberrypi/firmware"
> +LICENSE="GPL-2 raspberrypi-videocore-bin"
> +SLOT="0"
> +RESTRICT="binchecks strip"
> +
> +RDEPEND="sys-boot/raspberrypi-firmware"
> +
> +if [[ "${PV}" == 9999 ]]; then
> 
> What's the purpose in a non 9999 ebuild?

Explained in https://github.com/gentoo/gentoo/pull/11465#issuecomment-478381297

**Comment by andrey-utkin** [comment]
On Sat, Mar 30, 2019 at 05:14:55AM -0700, Conrad Kostecki wrote:
> ConiKost commented on this pull request.
> 
> 
> 
> > @@ -0,0 +1,33 @@
> +# Copyright 1999-2019 Gentoo Authors
> +# Distributed under the terms of the GNU General Public License v2
> +
> +EAPI=7
> +
> +inherit mount-boot
> +
> +DESCRIPTION="Raspberry Pi (all versions) kernel and modules"
> +HOMEPAGE="https://github.com/raspberrypi/firmware"
> +LICENSE="GPL-2 raspberrypi-videocore-bin"
> +SLOT="0"
> +RESTRICT="binchecks strip"
> 
> Why strip? It's allowed and other devs also said on my other packges, that strip doesn't count as "do not modify". I also would drop binchecks, as no QA errors show up.

It was this way. I kept it this way.
Does it cause any problem? I am not aware of any.

I do see where from you are coming with your change. Indeed `ebuild(5)`
says
```
binchecks
       Disable all QA checks for binaries.  This should ONLY be used in packages
       for which binary checks make no sense (linux-headers and kernel-sources, for
       example, can safely be skipped since they have no binaries).  If the binary
       checks need to be skipped for other reasons (such as proprietary binaries),
       see the QA CONTROL VARIABLES section for more specific exemptions.
```

But I just haven't had enough time to research this enough to ensure
this won't break it.

I am happy to disband this `RESTRICT` setting if we ensure the
correctness of replacement measures.

**Comment by ConiKost** [comment]
> Does it cause any problem? I am not aware of any.

Just saying, since on all of my previous PRs with binary license, this got nit pick on ```strip``` by other gentoo devs.

> But I just haven't had enough time to research this enough to ensure this won't break it.

Worst case would be QA errors, but on my tests, I didn't got any with removing this.

**Comment by andrey-utkin** [comment]
On Sun, Mar 31, 2019 at 11:01:05PM +0000, Conrad Kostecki wrote:
> > Does it cause any problem? I am not aware of any.
> 
> Just saying, since on all of my previous PRs with binary license, this got nit pick on ```strip``` by other gentoo devs.

I replied elsewhere.

Just collect the arguments/justification you was given by other devs,
and put them into commit message.

> 
> > But I just haven't had enough time to research this enough to ensure this won't break it.
> 
> Worst case would be QA errors, but on my tests, I didn't got any with removing this.

Worst case I am afraid would be broken binaries.
Possibly my fears are unjustified. But that's the point - explain the
change, demonstrate that the change is made competently, and everybody
will be warm and fuzzy with your changes.

**Comment by ConiKost** [comment]
> Worst case I am afraid would be broken binaries.

Sorry, but I don't get this. How can be binaries broken? QA only does some checks, but non of them could make them broke. The could be only broken already by upstream? Anyway, as I said, no errors for me without QA, so this should be fine in this case.

**Comment by andrey-utkin** [comment]
On Mon, Apr 01, 2019 at 09:59:37AM +0000, Conrad Kostecki wrote:
> > Worst case I am afraid would be broken binaries.
> 
> Sorry, but I don't get this. How can be binaries broken? QA only does some checks, but non of them could make them broke. The could be only broken already by upstream? Anyway, as I said, no errors for me without QA, so this should be fine in this case.

Stripping modifies binaries which can, in theory, lead to unexpected results.
In your PR restriction of stripping was gone.

**Pull request closed without merging**